### PR TITLE
Security basics

### PR DESCRIPTION
**Pull request by mykolaskrypka**

### CODE REVIEW

File: src/main/java/mate/academy/dto/UserRegistrationRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import jakarta.validation.constraints.Size;
+import lombok.Data;
+import mate.academy.validation.FieldMatch;
+
+@Data
+@FieldMatch(first = "password", second = "repeatPassword",
+        message = "The password fields must match")
+public class UserRegistrationRequestDto {
+    @NotNull
+    private String email;
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/dto/UserRegistrationRequestDto.java]: Is it enough for validate email?

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/dto/UserRegistrationRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.Data;
+import mate.academy.validation.FieldMatch;
+import mate.academy.validation.ValidEmail;
+
+@Data
+@FieldMatch(first = "password", second = "repeatPassword",
+        message = "The password fields must match")
+public class UserRegistrationRequestDto {
+    @NotNull
+    @ValidEmail
```
Comment (Rommelua) [inline src/main/java/mate/academy/dto/UserRegistrationRequestDto.java]:
```suggestion
    @Email
```

File: src/main/java/mate/academy/dto/UserRegistrationRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package mate.academy.dto;
+
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import lombok.Data;
+import mate.academy.validation.FieldMatch;
+import mate.academy.validation.ValidEmail;
```
Comment (Rommelua) [inline src/main/java/mate/academy/dto/UserRegistrationRequestDto.java]:
```suggestion
import jakarta.validation.constraints.ValidEmail;
```

File: src/main/java/mate/academy/validation/EmailValidator.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package mate.academy.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+public class EmailValidator implements ConstraintValidator<ValidEmail, String> {
```
Comment (Rommelua) [inline src/main/java/mate/academy/validation/EmailValidator.java]: No reason for custom validation

File: src/main/java/mate/academy/validation/FieldMatchValidator.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
+package mate.academy.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import org.apache.commons.beanutils.PropertyUtils;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch, Object> {
```
Comment (Rommelua) [inline src/main/java/mate/academy/validation/FieldMatchValidator.java:7]:
```suggestion
public class FieldMatchValidator implements ConstraintValidator<FieldMatch, UserRegistrationRequestDto> {
```

**Pull request merged**